hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -114,6 +116,59 @@ public class MapConfig implements IdentifiedDataSerializable, NamedConfig {
     private boolean perEntryStatsEnabled = DEFAULT_ENTRY_STATS_ENABLED;
     private int backupCount = DEFAULT_BACKUP_COUNT;
     private int asyncBackupCount = MIN_BACKUP_COUNT;
+    private SyncBackupTimeoutPolicy syncBackupTimeoutPolicy = SyncBackupTimeoutPolicy.GLOBAL;
+
+    /**
+     * Specifies action takes when {@link
+     * com.hazelcast.spi.properties.ClusterProperty#OPERATION_BACKUP_TIMEOUT_MILLIS}
+     * elapses when waiting for completion of synchronous backups.
+     */
+    public enum SyncBackupTimeoutPolicy {
```

Comment:
Maybe instead of introducing a new config option, we can keep the new option internal and jet specific code can access map-proxy to set it.

If there will be any convincing demand from users in future, we can think to introduce a config option.

Reasoning is: Being conservative seems better when adding new api, we can add it later anytime but adding it without any public reason can be hard to explain and can increase complexity a step further. So we can think to keep it internal for now.

Reply:
I do not like casting to private classes, but there is a precedent for it: SQL already casts `IMap` to `MapProxyImpl` in some processors. So yes, it can be done, there will be less documentation to write and no client-protocol changes.

However, I still think that knowing that the IMap.put or other modifying operation was not replicated and can disappear can be useful. Maybe not very much when IMap is used as cache, but more when IMap is used as data storage.

Reply:
Keeping it private is a power, we can do whatever we want with that, maybe in future IMap will not be used instead a new solution will be implemented and it can be used instead of IMap. But playing with public api is painful in the same way. I think we should be opting to refrain from it as much as possible if there is no convincing user request against it. This is to control complexity for users and us.

Reply:
moved to private API. I will have to do:
```
((MapProxyImpl)map).setSyncBackupTimeoutPolicy(SyncBackupTimeoutPolicy.INDETERMINATE);
```
when Jet obtains IMap but we can live with that.